### PR TITLE
Add tests for auth module (Fixes #5)

### LINKED ISSUE #5: Add/improve tests for the authentication module
* Increase the coverage for existing tests (e.g., `src/thunderbird_accounts/authentication/tests.py`)
* Add tests for modules with zero coverage (such as `src/thunderbird_accounts/authentication/middleware.py`)

### PR DESCRIPTION
**Pull request by radishmouse**
Closes #5 

Adds additional tests for APITestCase
Adds tests for middleware (FXABackend and ClientSetAllowedHostsMiddleware)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -73,4 +73,18 @@ Ensure you have the requirements in docs installed and run the following command
 
 ```shell
 sphinx-build docs build
-```
\ No newline at end of file
+```
+
+## Running tests in docker
+
+Make sure that the containers are alrady running.
```
Comment (rwood-moz) [inline README.md:80]: nit: typo `alrady`

File: src/thunderbird_accounts/authentication/tests.py [hunk 2]
```diff
@@ -71,3 +116,87 @@ def test_handle_auth_callback_response_with_explicit_redirect_url(self):
...
+
+        assert user
+        self.assertEqual(user.uuid, self.user.uuid)
+        self.assertEqual(user.fxa_id, self.user.fxa_id)
+
+        # Confirm new email overwrites original when doing lookup by fxa id
+        self.assertEqual(user.email, 'new@test.com')
+
+    def test_authenticate_with_matching_email(self):
+        """Test authentication when email matches existing user, but fxa id is nwe"""
```
Comment (rwood-moz) [inline src/thunderbird_accounts/authentication/tests.py:142]: nit: typo `nwe`

File: src/thunderbird_accounts/authentication/tests.py [hunk 3]
```diff
@@ -71,3 +116,87 @@ def test_handle_auth_callback_response_with_explicit_redirect_url(self):
...
+
+    def test_authenticate_with_no_match(self):
+        """Test authentication with new user"""
+        user = self.backend.authenticate(self.request, 'new_fxa_id', 'new@test.com')
+
+        self.assertIsNone(user)
+
+    def test_get_user_with_valid_id(self):
+        user = self.backend.get_user(self.user.uuid)
+        self.assertIsNotNone(user)
```
Comment (rwood-moz) [inline src/thunderbird_accounts/authentication/tests.py:160]: Maybe also assert on one of the user attributes? Just a suggestion :)

File: src/thunderbird_accounts/authentication/tests.py [hunk 4]
```diff
@@ -1,32 +1,77 @@
...
 from thunderbird_accounts.authentication.utils import handle_auth_callback_response
 from thunderbird_accounts.client.models import Client, ClientEnvironment
 
+from thunderbird_accounts.authentication.middleware import FXABackend, ClientSetAllowedHostsMiddleware
+
 
 class APITestCase(TestCase):
     def setUp(self):
+        self.GET_LOGIN_PATH = 'http://testserver/api/v1/auth/get-login/'
+        self.BAD_CREDENTIALS_MSG = 'Authentication credentials were not provided.'
```
Comment (rwood-moz) [inline src/thunderbird_accounts/authentication/tests.py:21]: nit: Maybe put these in a separate `const.py` file, then the might be easier to find if they change? Although there's only the two values so no biggie!

**Review by rwood-moz — COMMENTED** [review]
Cool! Just provided some tiny feedback (I didn't actually run the tests though as I don't have thunderbird-account dev env setup locally).

File: src/thunderbird_accounts/authentication/tests.py (hunk 3 above)
Comment (radishmouse) [reply src/thunderbird_accounts/authentication/tests.py:160]: Ah! Excellent idea

File: src/thunderbird_accounts/authentication/tests.py (hunk 4 above)
Comment (radishmouse) [reply src/thunderbird_accounts/authentication/tests.py:21]: I think I might hold off for the moment? I wonder if these values are specific enough to the test that they could stay for now.

File: README.md [hunk 5]
```diff
@@ -73,4 +73,18 @@ Ensure you have the requirements in docs installed and run the following command
...
+
+Make sure that the containers are already running.
+
+To run all tests:
+```sh
+docker compose exec backend uv run python manage.py test
+```
+
+To run tests for a specific module:
+```sh
```
Comment (MelissaAutumn) [inline README.md:88]: Of you use sh here, can you switch to sh within the document?

File: src/thunderbird_accounts/authentication/tests.py (hunk 4 above)
Comment (MelissaAutumn) [reply src/thunderbird_accounts/authentication/tests.py:21]: I'm fine with these being test class level...at least until we use them in more than one test class 🤔

File: src/thunderbird_accounts/settings.py [hunk 6]
```diff
@@ -26,8 +26,10 @@
 IS_TEST = 'test' in sys.argv
 
 if DEBUG:
-    load_dotenv()
-
+    if 'test' in sys.argv:
```
Comment (MelissaAutumn) [inline src/thunderbird_accounts/settings.py:29]:
```suggestion
    if IS_TEST:
```

It's defined just above there. 😄

File: Dockerfile [hunk 7]
```diff
@@ -21,7 +21,9 @@ COPY scripts/dev-entry.sh scripts/dev-entry.sh
 
 # Dev only
 RUN echo '!! If the next step fails, copy .env.example to .env in the backend folder !!'
-COPY .env .
+
+COPY .env* .
```
Comment (MelissaAutumn) [inline Dockerfile:25]: I'm okay just copying over .env and .env.test individually. But this works too.

**Review by MelissaAutumn — APPROVED** [review]
Looks good, just a few nits.

These might change a little bit as there's some planned infra changes on how accounts works. But thank you!

File: README.md (hunk 5 above)
Comment (radishmouse) [reply README.md:88]: Ah! I'll switch it to `shell` for the new snippets since that's what was already in the doc.

File: Dockerfile (hunk 7 above)
Comment (radishmouse) [reply Dockerfile:25]: What are the chances there will be other ones like `.env.prod` and `.env.stage`?

Comment (MelissaAutumn) [reply Dockerfile:25]: None until there is 😎
